pkg/controller/job/job_controller.go

```diff
@@ -1292,15 +1306,33 @@ func getStatus(job *batch.Job, pods []*v1.Pod, uncounted *uncountedTerminatedPod
 		return p.Status.Phase == v1.PodSucceeded
 	}))
 	failed += int32(countValidPodsWithFilter(job, pods, uncounted.Failed(), expectedRmFinalizers, func(p *v1.Pod) bool {
-		if p.Status.Phase == v1.PodFailed {
-			return true
+		if feature.DefaultFeatureGate.Enabled(features.JobPodFailurePolicy) {
+			isFailed := p.Status.Phase == v1.PodFailed
+			if !isFailed {
+				// When tracking with finalizers: counting deleted Pods as failures to
```

Comment:
This might overcomplicate things, but if PodDisruptionConditions is enabled, then we don't need this conditional.

Reply:
I was thinking about it, but it seems like a micro-ptimization which is not relevant in the long run when both gates disappear. 

Also, users are free to introduce custom conditions and handle them in the job controller by enabling just `JobPodFailurePolicy`.

Reply:
but this is about the phase set to `Failed`, which is specific to the garbage collector.

In the long run, when both features are enabled, we want to just rely on the phase.

Reply:
I see, there is another place in job_controller which should be simplified: https://github.com/kubernetes/kubernetes/blob/4f74844299489e5edd2b1feda4476399dfced62d/pkg/controller/job/job_controller.go#L1022
Maybe this calls for a separate PR for Beta?

Reply:
yes, that works for me

Reply:
Ok, updated the appropriate note in KEP